lib/smart_proxy_reports/spooled_http_client.rb

```diff
@@ -52,37 +52,50 @@ def stop
       @worker.join
     end
 
+    def get_endpoint(name)
+      return "/api/v2/host_reports" if name[name.index("-") + 1] == "r"
+      return "/api/v2/hosts/facts"
+    end
+
+    def connect(http, factory)
+      processed = 0
+      files = Dir.glob(spool_path("todo", "*")).sort
+      files.each do |filename|
```

Comment:
You can do this on a single line:

    Dir.glob(spool_path("todo", "*")).sort.each do ...

Reply:
Can you drop a comment before line 62:

On most Linux filesystems, glob is already sorted. Spool order must be kept, therefore we call quick sort to ensure all files are in order.